### PR TITLE
Update IBC smart contract to reflect lvl3 finality digest changes

### PR DESCRIPTION
**Pull request by systemzax**

### CODE REVIEW

File: unittests/svnn_ibc_tests.cpp [hunk 1]
```diff
@@ -503,7 +494,7 @@ BOOST_AUTO_TEST_SUITE(svnn_ibc)
                   ("major_version", 1)
                   ("minor_version", 0)
                   ("finalizer_policy_generation", 1)
-                  ("final_on_strong_qc_block_num", 14)
+                  //("final_on_strong_qc_block_num", 14)
```
Comment (linh2931) [inline unittests/svnn_ibc_tests.cpp]: Remove this commented out code.

File: unittests/test-contracts/savanna/common/savanna.hpp [hunk 2]
```diff
@@ -292,53 +296,38 @@ namespace savanna {
...
-
       //if a new finalizer policy is promoted to last pending status, it could (but is not guaranteed to) become active.
       //Therefore, we provide a mechanism to include finalizer policies into a proof of finality.
       //This allows the contract to obtain knowledge about them and to record them in its internal state.
       std::optional<finalizer_policy_input> new_finalizer_policy;
-
-      std::optional<checksum256> reversible_blocks_mroot;
-
       std::optional<uint32_t> last_pending_finalizer_policy_start_num;
+      //std::optional<checksum256> level_3_commitments_digest;
```
Comment (linh2931) [inline unittests/test-contracts/savanna/common/savanna.hpp:304]: Remove this line.

File: unittests/test-contracts/savanna/common/savanna.hpp [hunk 3]
```diff
@@ -292,53 +296,38 @@ namespace savanna {
...
-            });
-
-            checksum256 l3_digest = sha256(l3_packed.data(), l3_packed.size());
-
             auto l2_packed = eosio::pack(level_2_commitments_t{
                .last_pending_fin_pol_digest  = policy_digest, 
-               .last_pending_fin_pol_start_num = resolved_last_pending_finalizer_policy_start_num,
-               .l3_commitments_digest = l3_digest
+               .last_pending_fin_pol_start_num =  last_pending_finalizer_policy_start_num.value(),
+               .l3_commitments_digest = witness_hash
```
Comment (linh2931) [inline unittests/test-contracts/savanna/common/savanna.hpp:323]: Is `witness_hash` the same as l3_commitments?

Comment (systemzax) [reply unittests/test-contracts/savanna/common/savanna.hpp:328]: Sometimes.

I added a comment to explain it.

**Review by greg7mdp — APPROVED** [review]

**Comment by ericpassmore** [comment]
Note:start
group: STABILITY
category: TEST
summary: Update IBC smart contract to reflect finality digest changes.
Note:end

**Pull request merged**